index.js

```diff
@@ -0,0 +1,292 @@
+            let local = new Date(this);
+            local.setMinutes(this.getMinutes() - this.getTimezoneOffset());
+            return local.toJSON().slice(0,10);
+        
+        
+        });
+
+        let dateNow = new Date().toDateInputValue();
+        dateNow = dateNow.replace("-" , '')
+        dateNow = dateNow.replace("-",'')
+        dateNow = Number(dateNow)
+
+        let Idate = note_date
+        Idate = Idate.replace("-",'')
+        Idate = Idate.replace("-",'')
+        Idate = Number(Idate)
+      
+         if(Idate < dateNow){ return false;}
+    }
+   // local storage functions 
+
+    function saveToLocalStorage(key, value) {
+        localStorage.setItem(key, JSON.stringify(value));
+      }
+
+      function init() {
+        arrayOfNotes = JSON.parse(localStorage.getItem("stickyNotes")) || [];
+        draw(arrayOfNotes);
+      }
+      init();
+
+function sortBy(value,arr) {
+    switch(value){
+        case ("All"):
+            draw(arr);
+            break;
+        case ("ToDo"):
+            sortToDo(arr)
+            break;
+        case ("Today"):
+            sortToday(arr) 
+            break;
+        case ("Completed"):
+            sortCompleted(arr)
+            break;
+            
+    }
+}
+
+function sortToday(arr) {
+    Date.prototype.toDateInputValue = (function() {
+        let local = new Date(this);
+     
+        local.setMinutes(this.getMinutes() - this.getTimezoneOffset());
+        return local.toJSON().slice(0,10);
+    });   
+    clearBoard()
+    for (let index = 0; index < arr.length; index++) {
+        let dateNow = new Date().toDateInputValue();
+let dateYear = dateNow.slice(0, 4);
+let dateMonth = dateNow.slice(5, 7);
+let dateDay = dateNow.slice(8, 10);
+dateNow = (dateDay + dateMonth + dateYear)
+        dateNow = Number(dateNow)
+    console.log(dateNow)
+       let currentNote = arr[index]
+        let Idate = currentNote.note_date
+    Idate = Idate.replace("-",'')
+    Idate = Idate.replace("-",'')
+    Idate = Number(Idate)
+    console.log(Idate)
+     if(Idate === dateNow ){
+         
+        drawNote(arr[index])
+     }
+     else{clearBoard()}}
+     
+}
+
+function sortCompleted(arr) {
+    clearBoard()
+    for (let index = 0; index < arr.length; index++) {
+        let currentNote = arr[index]
+        let completed = currentNote.completed
+        console.log(completed)
+        if( completed === true) {
+            
+            console.log(arr[index])
+            drawNote(arr[index])
+            
+        }
+    }
+}
+
+function sortToDo(arr) {
+    clearBoard()
+    for (let index = 0; index < arr.length; index++) {
+        let currentNote = arr[index]
+        let completed = currentNote.completed
+        console.log(completed)
+        if( completed === false) {
+            
+            console.log(arr[index])
+            drawNote(arr[index])
+            
+        }
+    }
+}
+//DATE NOW GENERATOR
+Date.prototype.toDateInputValue = (function() {
```

Comment:
In general it is a bad practice to extend base types because if you use other thirdparty JS (which you always use, like jQuery or other framework) you can mess their code.

You don't have to use also this kind of extension, you can build a function called toDateInputValue that receives a parameter date and if you replace all the "this" in the function with date it will work perfectly.